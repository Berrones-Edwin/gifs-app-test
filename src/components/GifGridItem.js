import Proptypes from 'prop-types'
import React from 'react'

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={title} />
      <p> {title} </p>
    </div>
  )
}

GifGridItem.prototypes = {
  title: Proptypes.string.isRequired,
  url: Proptypes.string.isRequired
}
