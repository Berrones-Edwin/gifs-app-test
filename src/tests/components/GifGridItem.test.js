import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import React from 'react'
import { GifGridItem } from '../../components/GifGridItem'

describe('Test component <GifGridItem/>', () => {
  let title = ''
  let url = ''
  let wrapper = shallow(<GifGridItem title={title} url={url} />)
  beforeEach(() => {
    title = 'My title'
    url =
      'https://images.pexels.com/photos/10162332/pexels-photo-10162332.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940'
    wrapper = shallow(<GifGridItem title={title} url={url} />)
  })

  test('should component in screen', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('Should image has url and alt', () => {
    const img = wrapper.find('img')
    const { src, alt } = img.props()
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('should componet has a title', () => {
    const _title = wrapper.find('p').text().trim()

    expect(_title).toBe(title)
  })

  test('should has the class animate__fadeIn', () => {
    const div = wrapper.find('div')
    const hasClassDiv = div.props().className.includes('animate__fadeIn')
    expect(hasClassDiv).toBe(true)
  })
})
