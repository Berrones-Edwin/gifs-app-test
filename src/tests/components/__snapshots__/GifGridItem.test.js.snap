// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component <GifGridItem/> should component in screen 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="My title"
    src="https://images.pexels.com/photos/10162332/pexels-photo-10162332.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940"
  />
  <p>
     
    My title
     
  </p>
</div>
`;
